detect-python-version/action.yml

```
@@ -0,0 +1,42 @@
+name: Determine Python and Poetry versions
+description: Determine Python and Poetry versions
+
+inputs:
+  python_version:
+    description: "Python version to use"
+    required: false
+  working_directory:
+    description: "Working directory, defaults to GITHUB_WORKSPACE"
+    required: false
+    default: ${{ github.workspace }}
+
+outputs:
+  python-major-version:
+    description: "Python version"
+    value: ${{ steps.detect-versions.outputs.python-major-version }}
+  poetry-version:
+    description: "Poetry version"
+    value: ${{ steps.detect-versions.outputs.poetry-version }}
+
+runs:
+  using: composite
+  steps:
+    # Empty arrays cannot be provided with `-f`, see https://github.com/cli/cli/issues/1484
+    - id: detect-versions
+      run: |
+        if [ -z "${{ inputs.python_version }}" ] ; then
+            PYTHON_VERSION="$(sed -n -e '/^\[metadata\]/,/^\[/p' poetry.lock | sed -n -e 's/^python-versions[[:space:]]*=[[:space:]]*//p' | tr -d \"'[:space:]'\'~^)"
+        else
+            PYTHON_VERSION="${{ inputs.python_version }}"
+        fi
+
+        if [ -z "${{ inputs.poetry_version }}" ] ; then
+            POETRY_VERSION="$(head --lines=1 poetry.lock | grep --only-matching '[0-9]\+\.[0-9]\+\.[0-9]\+')"
+        else
+            POETRY_VERIONS="${{ inputs.poetry_version }}"
```

I think this typo is already on master, please fixit here.
```suggestion
            POETRY_VERSION="${{ inputs.poetry_version }}"
```